src/test/java/com/google/firebase/database/util/EmulatorHelperTest.java

```diff
@@ -0,0 +1,144 @@
+        new CustomTestCase("http://my-custom-hosted-emulator.com:80?ns=dummy-ns",
+            "http://localhost:8080/ns=ns-2",
+            "http://my-custom-hosted-emulator.com:80/?ns=dummy-ns"),
+        new CustomTestCase("http://localhost:9000?ns=ns-1", "localhost:8080",
+            "http://localhost:9000/?ns=ns-1"),
+        new CustomTestCase("http://localhost:9000?ns=ns-1", "http://localhost:8080/ns=ns-2",
+            "http://localhost:9000/?ns=ns-1"),
+        new CustomTestCase("http://localhost:9000/a/b/c?ns=ns-1", "http://localhost:8080/ns=ns-2",
+            "http://localhost:9000/a/b/c/?ns=ns-1"),
+
+        // cases where the supplied DB URL is not an emulator URL, so we extract ns from it
+        // and append it to the emulator URL from env var(if it is valid)
+        new CustomTestCase("https://valid-namespace.firebaseio.com", "localhost:8080",
+            "http://localhost:8080/?ns=valid-namespace"),
+        new CustomTestCase("https://valid-namespace.firebaseio.com", "custom-emulator-url:90",
+            "http://custom-emulator-url:90/?ns=valid-namespace"),
+        new CustomTestCase("https://valid-namespace.firebaseio.com/a/b/c", "custom-emulator-url:90",
+            "http://custom-emulator-url:90/a/b/c/?ns=valid-namespace"),
+        new CustomTestCase("https://valid-namespace.firebaseio.com", "192.123.212.145:90",
+            "http://192.123.212.145:90/?ns=valid-namespace"),
+        new CustomTestCase("https://valid-namespace.firebaseio.com", "[::1]:90",
+            "http://[::1]:90/?ns=valid-namespace"),
+        new CustomTestCase("https://firebaseio.com?ns=valid-namespace", "localhost:90",
+            "http://localhost:90/?ns=valid-namespace"),
+        new CustomTestCase(null, "localhost:90", "http://localhost:90/?ns=default"),
+        new CustomTestCase("", "localhost:90", "http://localhost:90/?ns=default")
+    );
+
+    for (CustomTestCase tc : testCases) {
+      TestUtils.setEnvironmentVariables(
+          ImmutableMap.of(EmulatorHelper.FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR,
+              Strings.nullToEmpty(tc.envVariableUrl)));
+      assertEquals(tc.expectedEmulatorUrl, EmulatorHelper.getEmulatorUrl(tc.suppliedDbUrl));
+      TestUtils.unsetEnvironmentVariables(
+          ImmutableSet.of(EmulatorHelper.FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR));
+    }
+  }
+
+  @Test
+  public void testInvalidEmulatorUrlFromEnvVarThrows() {
+    List<String> invalidEnvVars = ImmutableList.of(
+        "http://localhost:8080",
+        "http://localhost:8080?ns=test-ns",
+        "localhost"
+    );
+    for (String invalidEnvVar : invalidEnvVars) {
+      TestUtils.setEnvironmentVariables(
+          ImmutableMap.of(EmulatorHelper.FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR,
+              Strings.nullToEmpty(invalidEnvVar)));
+      thrown.expect(IllegalArgumentException.class);
+      EmulatorHelper.getEmulatorUrl("https://valid-namespace.firebaseio.com");
+      TestUtils.unsetEnvironmentVariables(
```

Comment:
Wrap this in a try/finally so that `unsetEnvironmentVariables` is called.

Reply:
Removed this code, so resolving the comment.